### PR TITLE
vue-components: create TextInput component

### PR DESCRIPTION
**Pull request by jakobw**
# TextInput component :sparkles: 

- [x] initial styling
- [x] width prop
- [x] responsiveness
- [x] label
- [x] placeholder
- [x] disabled
- [x] states (hover, active/focus, ...)
- [x] add warning/error icons

Phabricator task: https://phabricator.wikimedia.org/T263734
Figma spec: https://www.figma.com/file/gi3JYwSUQYhKS0mHwK25cQwi/WiKit?node-id=3975%3A1

Most of the code here was written in pair programming by @guergana and @jakobw.

### CODE REVIEW

File: vue-components/src/components/TextInput.vue [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+			/>
+			{{ error.message }}
+		</div>
+	</label>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import Icon from './Icon.vue';
+
```
Comment (SaiSan-WMDE) [inline vue-components/src/components/TextInput.vue:26]: /* eslint-disable no-trailing-spaces */
/**
 *  Text input fields are form elements that let users input and edit values in the form of text.
 */
/* eslint-enable no-trailing-spaces */
```suggestion

```

Comment (SaiSan-WMDE) [inline vue-components/src/components/TextInput.vue:26]:
```suggestion
/* eslint-disable no-trailing-spaces */
/**
 *  Text input fields are form elements that let users input and edit values in the form of text.
 */
/* eslint-enable no-trailing-spaces */
```

Comment (SaiSan-WMDE) [reply vue-components/src/components/TextInput.vue:26]: Just realized we hadn't added this component's description. Is this all that needs to be done for the description to be displayed in the Docs page?

Comment (guergana) [reply vue-components/src/components/TextInput.vue:26]: Yes, the comment at the top is the comment that appears in the docs tab above the rendered element. :)

Comment (jakobw) [reply vue-components/src/components/TextInput.vue:26]: Thanks, @SaiSan-WMDE I added the comment with the latest update.

**Comment by jakobw** [comment]
The latest update is only a rebase on top of the latest (small) changes that happened in #197 .

**Review by guergana — APPROVED** [review]
Looks good plus it has been tested and discussed with design team.

**Pull request merged**